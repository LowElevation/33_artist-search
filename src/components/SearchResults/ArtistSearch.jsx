import React from "react";
import PropTypes from "prop-types";
import uuid from "react-uuid";
import { Link } from "react-router-dom";

const ArtistSearch = ({ artists }) => {
  const artistList = artists.map((artist) => {
    return <li key={uuid()}>
      <Link to={`/releases/${artist.id}`}>
        {artist.name}
      </Link>
    </li>;
  });
  return <ul>{artistList}</ul>;
};

ArtistSearch.propTypes = {
  artists: PropTypes.array,
};

export default ArtistSearch;
