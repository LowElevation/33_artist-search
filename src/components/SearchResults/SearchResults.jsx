import React from "react";
import PropTypes from "prop-types";
import uuid from "react-uuid";
import { Link } from "react-router-dom";

const SearchResults = ({ artists }) => {
  const artistList = artists.map((artist) => {
    return <li key={uuid()}>
      <Link to={`/releases/${artist.id}`}>
        {artist.name}
      </Link>
    </li>;
  });
  return <ul>{artistList}</ul>;
};

SearchResults.propTypes = {
  artists: PropTypes.array,
};

export default SearchResults;
