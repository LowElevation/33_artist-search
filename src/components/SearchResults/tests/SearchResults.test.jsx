import React from "react";
import { render, cleanup } from "@testing-library/react";
import SearchResults from "../SearchResults";

describe("SearchResults component", () => {
  afterEach(() => cleanup());
  it("renders SearchResults", () => {
    const { asFragment } = render(<SearchResults artists={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
