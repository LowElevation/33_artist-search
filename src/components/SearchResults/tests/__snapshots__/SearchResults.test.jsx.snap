// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders SearchResults 1`] = `
<DocumentFragment>
  <ul>
    Go
  </ul>
</DocumentFragment>
`;
